internal/core/items.rs

```diff
@@ -424,13 +425,48 @@ pub struct TouchArea {
     pub mouse_y: Property<LogicalLength>,
     pub mouse_cursor: Property<MouseCursor>,
     pub clicked: Callback<VoidArg>,
+    pub double_clicked: Callback<VoidArg>,
     pub moved: Callback<VoidArg>,
     pub pointer_event: Callback<PointerEventArg>,
     pub scroll_event: Callback<PointerScrollEventArg, EventResult>,
     /// FIXME: remove this
     pub cached_rendering_data: CachedRenderingData,
     /// true when we are currently grabbing the mouse
     grabbed: Cell<bool>,
+    clicked_timer: Timer,
+}
+
+impl TouchArea {
+    fn start_clicked_timer(self: Pin<&Self>, item_rc: &ItemRc) {
+        let max_click_interval = crate::platform::PLATFORM_INSTANCE
+            .with(|p| p.get().map(|p| p.click_interval()))
+            .unwrap_or_default();
+        let weak = item_rc.downgrade();
+
+        self.clicked_timer.start(
+            crate::timers::TimerMode::SingleShot,
+            max_click_interval,
+            move || {
+                let Some(item) = weak.upgrade() else {
+                    return;
+                };
+                let Some(touch_area) = item.downcast::<TouchArea>() else {
+                    return;
+                };
+                if Self::FIELD_OFFSETS.pressed.apply_pin(touch_area.as_pin_ref()).get() == false {
+                    Self::FIELD_OFFSETS.clicked.apply_pin(touch_area.as_pin_ref()).call(&());
+                }
+            },
+        );
+        assert!(self.clicked_timer.running());
+    }
+
+    fn stop_clicked_timer(self: Pin<&Self>) -> bool {
+        let result = self.clicked_timer.running();
+        self.clicked_timer.stop();
+        assert!(!self.clicked_timer.running());
```

Comment:
```suggestion
        debug_assert!(!self.clicked_timer.running());
```

That said, I don't like the idea of asserting here. In the event of this happening the user application crashes, which really doesn't do anyone a favor. If this is supposed to "verify" that our Time behaves, then the right thing is to do that in a unit test (which you did), so I'd remove this TBH and rely on Timer working (it's tested elsewhere, not in production code paths).